pkg/gofr/crud_handlers.go

```diff
@@ -86,13 +89,52 @@ func scanEntity(object interface{}) (*entity, error) {
 	tableName := getTableName(object, structName)
 	restPath := getRestPath(object, structName)
 
-	return &entity{
-		name:       structName,
-		entityType: entityType,
-		primaryKey: primaryKeyFieldName,
-		tableName:  tableName,
-		restPath:   restPath,
-	}, nil
+	e := &entity{
+		name:        structName,
+		entityType:  entityType,
+		primaryKey:  primaryKeyFieldName,
+		tableName:   tableName,
+		restPath:    restPath,
+		constraints: make(map[string]sql.FieldConstraints),
+	}
+
+	for i := 0; i < entityType.NumField(); i++ {
+		field := entityType.Field(i)
+		fieldName := toSnakeCase(field.Name)
+
+		sqlTag := field.Tag.Get("sql")
+		if sqlTag != "" {
+			constraints, err := parseSQLTag(sqlTag)
+			if err != nil {
+				return nil, err
+			}
+
+			e.constraints[fieldName] = constraints
+		}
+	}
+
+	return e, nil
+}
+
+func parseSQLTag(tag string) (sql.FieldConstraints, error) {
+	var constraints sql.FieldConstraints
+
+	tags := strings.Split(tag, ",")
+
+	for _, t := range tags {
```

Comment:
I would use `tag`

```suggestion
	for _, tag := range tags {
```

`t` is connoted for` *testing.T`

Reply:
sure renamed to tag